packages/frontend/src/pages/AttesterPage.jsx

```diff
@@ -19,12 +19,14 @@ export default observer(() => {
     const loadData = async () => {
       await Promise.all([
         unirep.loadAttesterDeployments(),
-        attester.loadEpochsByAttester(id),
-        attester.loadSignUpsByAttester(id),
-        attester.loadAttestationsByAttester(id),
-        attester.loadUSTByAttester(id),
+        attester.loadEpochsByAttester(attesterId),
+        attester.loadSignUpsByAttester(attesterId),
+        attester.loadAttestationsByAttester(attesterId),
+        attester.loadUSTByAttester(attesterId),
       ])
-      setCurrentEpoch(attester.epochsByAttester.length - 1)
+      setCurrentEpoch(
+        attester.epochsByAttester[attester.epochsByAttester.length - 1].number
+      )
```

Comment:
This code displays the attester's "Epochs Processed" as the latest epoch listed in the database, is that the data we want here? Or are we looking for the last sealed epoch? I'm not really getting how the EpochEnded event works - what the Epoch db table is showing me. I have epoch#s [3, 4, 5, 31, 32, 33, 35, 36, 304, 305, 306, 307, 762, 763, 3219, 3220, 3221, 3222, 3223]..
Some of these had signups or attestations during the epoch but some did not. Why are these added to the table and not others? 

--this shouldn't still be called `currentEpoch` here, I forgot to change it when I moved the calculation to `EpochView.jsx`

Reply:
Right, the `EpochEnded` event is emitted only when there is an action in the next epoch. The event emitted will be for the previous epoch. Any epoch with attestations is guaranteed to emit an `EpochEnded` (because it will be sealed after). This is complicated to reason about though because an epoch might be sealed many epochs later (which will cause a weird `EpochEnded` event). This event might get removed in the future.

The "Epochs Processed" should probably be the latest epoch a user can transition to. This is the latest epoch if the last epoch was sealed. If there is an epoch that has attestations but is not sealed, then it is the epoch before that. For now I think just latest epoch is fine.

I think maybe we should instead show the current epoch and maybe a red or green indicator showing if the epoch can be transitioned to, or if a previous epoch is awaiting sealing. This can be done in a future pr though.

Reply:
✅ this is addressed in #25